resources/js/Components/Form/DmEntryCreateForm.tsx

```diff
@@ -0,0 +1,332 @@
+import {useForm} from '@inertiajs/inertia-react'
+import AdapterDateFns from '@mui/lab/AdapterDateFns'
+import DatePicker from '@mui/lab/DatePicker'
+import LocalizationProvider from '@mui/lab/LocalizationProvider'
+import {
+    Box,
+    Button,
+    Chip,
+    Grid,
+    MenuItem,
+    Step,
+    StepLabel,
+    Stepper,
+    TextField,
+    Typography,
+} from '@mui/material'
+import {ErrorText, Link} from 'Components'
+import React, {useState} from 'react'
+import styled from 'styled-components'
+import {adventureType} from 'Types/adventure-data'
+import {EntriesData} from 'Types/entries-data'
+import {ItemData} from 'Types/item-data'
+import route from 'ziggy-js'
+
+import ItemCreateForm from './ItemCreateForm'
+
+type DmEntryCreateFormPropType = {
+    type: 'Edit' | 'Create'
+    onCloseDrawer?: () => void
+    editData?: EntriesData
+    editId?: number
+    user_id?: number
+    adventures?: adventureType[]
```

Comment:
i think adventures should be required no?

Reply:
Well we left them nullable because when we import from DnDBeyond we have no adventure data and the user has to fill that out later, so it sadly needs to be nullable

Reply:
make sense, can we have an initial value for it tho, if it's nullable just return `[]` perhaps?